### DISCUSSION TITLE
Specifying the Server Federation Admin URL

### DISCUSSION THREAD

**Discussion by dmoiler** [Esri/arcgis-powershell-dsc#514] [Q&A]
Hi All,

HNY 🍾 🎆 

I'm wondering, is there a way to specify the admin federation URL for Server? 

Configuring a web adaptor for service and web adaptor for administration seems like a neat solution, so if possible I'd like to specify both parameters rather than directly accessing the server via 6443.

![image](https://github.com/Esri/arcgis-powershell-dsc/assets/15136197/51fc7651-2b44-4bbd-a022-49e882e19585)

The server context seems to be used for the services URL from the deployments I've performed so far. 

Is there a way to specify that admin URL also?

![image](https://github.com/Esri/arcgis-powershell-dsc/assets/15136197/8c832aec-8eb2-4d4f-bc16-8a1049e29bbd)

Thanks!
Dean

**Comment by cameronkroeker** [comment]
Hi @dmoiler,

In your case the reason the ServerContext is only being used for the federated services url is because AdminAccessEnabled is set to false. When AdminAccessEnabled is set to true the module will use the ServerContext for both the services url and admin url.

Having two different web adaptors for federation is not currently supported by the module. However, one way that it might be possible would be to use ConfigData.Server.InternalLoadBalancer attribute (set to the FQDN of the WA, and ConfigData.Server.InternalLoadBalancerPort (set to 443). But this would require the WebAdaptorConfig.Context set to `arcgis`, instead of `agsmapadmin`, with AdminAccessEnabled set to true.

Thanks,
Cameron K.

**Comment by dmoiler** [comment]
Hi @cameronkroeker,

Thanks for clarifying. Currently we're simply using 6443 for server administration and don't allow admin access via the service URL. I had though using a separate web adaptor could be a neater solution (after seeing examples in the 4.2 wiki), and could simplify if we were to horizontally scale any roles. We've got several other servers federated and registered to this web server so unfortunately the arcgis context would only get us so far. 

Ideally we'd have agsptladmin, agsimgadmin, agsgeoadmin, etc.. for different server roles, but can keep to 6443 

Is there any likelihood this capability would be added to the roadmap for future versions of DSC?

Thanks,
Dean
